src/pages/Dashboard/components/ReportTable/TableRow/TableRow.tsx

```diff
@@ -65,13 +65,18 @@ function TableRow({ skill, projections }: TableRowProps): JSX.Element {
   };
 
   let maxNeededRoles = 0;
-  const neededNeeded: boolean[] = [];
   projections.forEach(({ needed }, i) => {
     maxNeededRoles = Math.max(maxNeededRoles, needed.roles?.length || 0);
-    neededNeeded.push(
-      (needed.roles && needed.roles[i] && needed.roles[i].value > 0) || false,
-    );
   });
+  const notNeededRows = Array.from({ length: maxNeededRoles }).map(
```

Comment:
Since the logic for this is pretty complex, it might be nice to pull this logic out of the component and into a function to keep it isolated and simplify the component a bit.

Reply:
leaving as is - follows the logic that's used in the jsx